docs/spec.md

```diff
@@ -1880,6 +1880,7 @@ This reframing uses the following helper functions:
 
 * `lhs_shape(n, hw, c) = permute([n] + hw + [c], [input_batch_dimension] + input_spatial_dimensions + [input_feature_dimension])`.
 * `result_shape(n1, hw, c1) = permute([n1] + hw + [c1], [output_batch_dimension] + output_spatial_dimensions + [output_feature_dimension])`.
+* `permute([j0, j1, ..., jR-1], permutation)`, with `size(permutation) = R`, produces `[i0, i1, ..., iR-1]` such that `j[d] = i[permutation[d]] for all d in [0, R)`.
```

Comment:
Proposed rewording: 

```
`permute([j0, j1, ..., jR-1], permutation) = [i0, i1, ..., iR-1]` where `j[d] = i[permutation[d]]`.
```

  1. Saying `=` instead of "produces" is consistent with the rest of the spec.
  2. Given the overhead of saying "for all...", I think we should shorten this to the original formulation, which is consistent with the spec of stablehlo.transpose.

Reply:
Thanks!
Done.